Sparkle/SUTouchBarProvider.m

```diff
@@ -0,0 +1,94 @@
+//
+//  SUTouchBarProvider.m
+//  Sparkle
+//
+//  Created by Yuxin Wang on 05/01/2017.
+//  Copyright © 2017 Sparkle Project. All rights reserved.
+//
+
+#import "SUTouchBarProvider.h"
+
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101202
+
+@interface SUTouchBarProvider() <NSTouchBarDelegate>
+
+@property (copy) NSString* indentifier;
+@property NSMutableArray<NSTouchBarItem*> *touchBarItems;
+
+@end
+
+@implementation SUTouchBarProvider
+
+@synthesize indentifier;
+@synthesize touchBarItems;
+@synthesize touchBar;
+
+-(instancetype)initWithIdentifier:(NSString *)anIdentifier
+{
+    if (!(self = [super init]))
+        return self;
+    
+    indentifier = [NSString stringWithFormat:@"%@.%@", [NSBundle mainBundle].bundleIdentifier, anIdentifier];
```

Comment:
Main bundle identifier means system level unique. SPARKLE_BUNDLE_IDENTIFIER means application level unique. I'm not sure which is "globally-unique", but I think it is not really matters here.